### ISSUE TITLE
Collapse content has fixed height so that the element lost self-adaption in nested structure with react 18 StrictMode

### ISSUE THREAD

**Issue by Gu-Miao** [mantinedev/mantine#1622]
### What package has an issue

@mantine/core

### Describe the bug

I have a nested Collapse component. I want it to be uncontrol. I render it with routes data.

What I want is if the current url is **/data/materials/ad52**, the 'Data'(first level menu item) should be open, the 'Materials'(second level menu item) should be open and 'Geometry'(third level menu item) should be active.

![image](https://user-images.githubusercontent.com/28031505/173277880-1c233472-fea0-4b4b-8fbd-716c84a3ab30.png)

I make this come ture but the issue appear when I refresh the browser. The menu items do be active and parents are open, but the second level menu item content get a fixed height so that a big blank space appear.

![Honeycam 2022-06-13 11-23-12](https://user-images.githubusercontent.com/28031505/173278115-6bc25bff-f77e-4eac-be22-6136115f7983.gif)

![GXCG{ `E$)C%{GCP3WY W6N](https://user-images.githubusercontent.com/28031505/173278216-e2b1ad86-906e-4158-b7d8-6a67822844b3.png)

I write this with `Collapse` component, and `Accordion` component has the same problem.

**And I find that if I remove `StrictMode`, it works fun.**

My network is not well when using codesandbox so that I provide my repo link below.

### In which browser did the problem occur

Latest chrome

### If possible, please include a link to a codesandbox with the reproduced problem

https://github.com/Gu-Miao/react-admin/tree/mantine-collapse-issue#react-admin

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

**Comment by rtivital** [comment]
I do not think that we should spend time fixing  strict mode issue, if it works fine without it, then just remove StrictMode and that's all.

**Comment by Gu-Miao** [comment]
Well, offical suggestion? If so, I will close this.

**Comment by rtivital** [comment]
I'm not really sure, what official means in this context – you definitely won't get arrested for using react strict mode, it would be just an inconvenience for you in dev mode since it does not impact production builds.
